core/src/main/java/io/grpc/util/RoundRobinLoadBalancer.java

```diff
@@ -88,9 +89,8 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
     List<EquivalentAddressGroup> servers = resolvedAddresses.getAddresses();
     Attributes attributes = resolvedAddresses.getAttributes();
     Set<EquivalentAddressGroup> currentAddrs = subchannels.keySet();
-    Set<EquivalentAddressGroup> latestAddrs = stripAttrs(servers);
-    Set<EquivalentAddressGroup> addedAddrs = setsDifference(latestAddrs, currentAddrs);
-    Set<EquivalentAddressGroup> removedAddrs = setsDifference(currentAddrs, latestAddrs);
+    Map<EquivalentAddressGroup, EquivalentAddressGroup> latestAddrs = stripAttrs(servers);
```

Comment:
`servers` may have repetitions as well as addresses with the same stripped address but different attributes. You may consider
```java
List<EquivalentAddressGroup> strippedServers = stripAttrs(servers);
<T> Set<T> setsDifference(Set<T> a, Collection<T> b);
```

And the identity of a subchannel is always determined by `originalAddressGroup`
```java
subchannels.put(originalAddressGroup, subchannel);
```

Reply:
Actually I think we shouldn't do `stripAttrs` at all.

Reply:
`stripAttrs` is to avoid calling Attributes.equals() via EAG.equals(). That's part of the reason the code is a bit wonky.